src/event-log/event-log-level.ts

```diff
@@ -76,46 +102,40 @@ export class EventLogLevel implements EventLog {
     return events;
   }
 
-  async deleteEventsByCid(tenant: string, cids: Array<string>): Promise<number> {
-    if (cids.length === 0) {
+  /**
+   * Deletes Events associated with a given cid. Deletes any indexes that were created along with it as well.
+   * @param tenant
+   * @param messageCids an array of messageCids to delete.
+   * @returns the number of events deleted.
+   */
+  async deleteEventsByCid(tenant: string, messageCids: Array<string>): Promise<number> {
+    if (messageCids.length === 0) {
       return 0;
     }
 
     const tenantEventLog = await this.db.partition(tenant);
     const cidLog = await tenantEventLog.partition(CIDS_SUBLEVEL_NAME);
+    const watermarkLog = await tenantEventLog.partition(WATERMARKS_SUBLEVEL_NAME);
 
-    let ops: LevelWrapperBatchOperation<string>[] = [];
-    const promises: Array<Promise<string | undefined>> = [];
-
-    for (const cid of cids) {
-      ops.push({ type: 'del', key: cid });
-
-      const promise = cidLog.get(cid).catch(e => e);
-      promises.push(promise);
-    }
-
-    await cidLog.batch(ops);
+    const ops: LevelWrapperBatchOperation<string>[] = [];
+    const cidOps: LevelWrapperBatchOperation<string>[] = [];
 
-    ops = [];
     let numEventsDeleted = 0;
-
-    const watermarks: Array<string | undefined> = await Promise.all(promises);
-    for (const watermark of watermarks) {
-      if (watermark) {
-        ops.push({ type: 'del', key: watermark });
-        numEventsDeleted += 1;
+    const indexDeletePromises: Promise<void>[] = [];
+    for (const messageCid of messageCids) {
+      const watermark = await cidLog.get(messageCid);
+      if (watermark === undefined) {
+        continue;
       }
+      ops.push({ type: 'del', key: watermark });
+      cidOps.push({ type: 'del', key: messageCid });
+      indexDeletePromises.push(this.index.delete(tenant, messageCid));
```

Comment:
Am I reading it correctly that we do not batch the deletion of indexes, but we delete the indexes one by one?

Reply:
The individual indexes within the index log for each message is batched, but the index.delete() method is what handles the lookup/deleting of the individual indexes.

So for each message we add a promise that will delete the index for the message.